tsdb/engine/tsm1/float.go

```diff
@@ -138,17 +138,17 @@ type FloatDecoder struct {
 	err error
 }
 
-func NewFloatDecoder(b []byte) (*FloatDecoder, error) {
+func NewFloatDecoder(b []byte) (FloatDecoder, error) {
 	// first byte is the compression type but we currently just have gorilla
 	// compression
 	br := bitstream.NewReader(bytes.NewReader(b[1:]))
 
 	v, err := br.ReadBits(64)
 	if err != nil {
-		return nil, err
+		return FloatDecoder{}, err
 	}
 
-	return &FloatDecoder{
+	return FloatDecoder{
```

Comment:
It just returns it by value so it'll copy the whole struct instead of a pointer to it. The caller can still mutate the struct since the methods are pointer receivers.